metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_v2.py

```diff
@@ -363,9 +363,17 @@ def get_upstreams_from_sql_parsing_result(
         for column_lineage in result.column_lineage or []:
             out_column = column_lineage.downstream.column
             for upstream_column_info in column_lineage.upstreams:
-                upstream_table_name = DatasetUrn.create_from_string(
+                upstream_table_id = DatasetUrn.create_from_string(
                     upstream_column_info.table
                 ).get_dataset_name()
+                if self.config.platform_instance and upstream_table_id.startswith(
```

Comment:
How does this happen? Perhaps we should separate platform instance out in the sql parser instead? cc @hsheth2

Reply:
imo the ideal outcome would be for the sql parser to respect `platform_instance` if one is passed in to the schema resolver

Reply:
I thought it already does that, but might be wrong

Reply:
So the sql lineage returned has
1. dataset urn
2. column to column lineage where each column is represented by (dataset urn, column_name)

What's missing here is that - we don't have raw dataset name (or fully qualified name) without platform instance in the result.

Reply:
@hsheth2 @asikowitz any thoughts on how to proceed here ?

Reply:
I think, this bug in snowflake view lineage can also be solved by converting sql parsing result to fine grained lineage directly instead of raw snowflake data models again. Let me attempt that.

Reply:
Done. @hsheth2 please check now.